ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -193,86 +282,56 @@ runDataDiffusion tracers
-        localErrorPolicy
-
-    runServer :: NetworkMutableState -> AddrInfo -> IO Void
-    runServer networkState address =
-      NodeToNode.withServer
-        (NetworkServerTracers
-          dtMuxTracer
-          dtHandshakeTracer
-          dtErrorPolicyTracer)
-        networkState
-        address
-        (daResponderApplication applications)
-        remoteErrorPolicy
-
-    runIpSubscriptionWorker :: NetworkMutableState -> IO Void
-    runIpSubscriptionWorker networkState = NodeToNode.ipSubscriptionWorker
-      (NetworkIPSubscriptionTracers
-        dtMuxTracer
-        dtHandshakeTracer
-        dtErrorPolicyTracer
-        dtIpSubscriptionTracer)
-      networkState
-      SubscriptionParams
-        { spLocalAddresses         = initiatorLocalAddresses
-        , spConnectionAttemptDelay = const Nothing
-        , spErrorPolicies          = remoteErrorPolicy
-        , spSubscriptionTarget     = daIpProducers
-        }
-      (daInitiatorApplication applications)
-
-    runDnsSubscriptionWorker :: NetworkMutableState -> DnsSubscriptionTarget -> IO Void
-    runDnsSubscriptionWorker networkState dnsProducer = NodeToNode.dnsSubscriptionWorker
-      (NetworkDNSSubscriptionTracers
-        dtMuxTracer
-        dtHandshakeTracer
-        dtErrorPolicyTracer
-        dtDnsSubscriptionTracer
-        dtDnsResolverTracer)
-      networkState
-      SubscriptionParams
-        { spLocalAddresses         = initiatorLocalAddresses
-        , spConnectionAttemptDelay = const Nothing
-        , spErrorPolicies          = remoteErrorPolicy
-        , spSubscriptionTarget     = dnsProducer
-        }
-      (daInitiatorApplication applications)
-
-
---
--- Auxilary functions
---
-
--- | Structural fold using 'Async.withAsync'.
---
-withAsyncs :: [IO a] -> ([Async.Async a] -> IO b) -> IO b
-withAsyncs as0 k = go [] as0
-  where
-    go threads []       = k threads
-    go threads (a : as) = Async.withAsync a $ \thread -> go (thread : threads) as
+    runIpSubscriptionWorker
+      :: Connections Connections.ConnectionId Socket.Socket Request (Concurrent.Reject reject) (Concurrent.Accept (ConnectionHandle IO))  IO
+      -> IO ()
+    runIpSubscriptionWorker connections =
+      case ipSubscriptionTargets (ispIps daIpProducers) initiatorLocalAddresses of
+        -- There were no addresses in the config (ispIps) which have an
+        -- address in initiatorLocalAddresses of the same family.
+        Nothing -> pure ()
+        Just connIds -> Subscription.worker
+          (contramap (WithIPList initiatorLocalAddresses (ispIps daIpProducers)) dtIpSubscriptionTracer)
+          dtErrorPolicyTracer
+          connIds
+          (ispValency daIpProducers)
+          ipRetryDelay
+          (\connId -> Socket.client connId IpSubscriptionConnection)
+          connections
+
+    -- FIXME probably not ideal that we make a resolver `mkResolverIO` for
+    -- each DNS target. Should make one (not have it take a port) and use it
```

Comment:
^ a typo, it should be 'take a part'

Reply:
No "port" is what I meant. Unclear language though. I meant to say, we should make it so that `mkResolverIO` does not hard-code one port for all resolutions. Rather, you should get back results which are IP addresses only. This way we could use one resolver for multiple different `dstPort`s.